### PR TITLE
Create pytorch objects inference fix

### PR DESCRIPTION
**Pull request by psfoley**
Credit to @sarthakpati for the base implementation!

## Proposed Changes
- Correctly populates header information for inference when only validation CSV is provided to `create_pytorch_objects` function
- Handles a couple other corner cases where training header information is needed
- Used in conjunction with FeTS Challenge 2022

## Checklist

- [ x ] I have read the [`CONTRIBUTING`](https://github.com/CBICA/GaNDLF/blob/master/CONTRIBUTING.md) guide
- [ x ] My PR is based from the [current GaNDLF master ](https://garygregory.wordpress.com/2016/11/10/how-to-catch-up-my-git-fork-to-master/)
- [ x ] Non-breaking change (would not break existing functionality): please provide as many details as possible for any breaking change 
- [ ] Function/class source code documentation added/updated
- [ ] Code has been [blacked](https://github.com/psf/black#usage) for style consistency
- [ ] If applicable, version information [has been updated in GANDLF/version.py](https://github.com/CBICA/GaNDLF/blob/master/GANDLF/version.py)
- [ ] If adding a git submodule, add to list of exceptions for black styling in [pyproject.toml](https://github.com/CBICA/GaNDLF/blob/master/pyproject.toml) file 
- [ ] [Usage documentation](https://github.com/CBICA/GaNDLF/blob/master/docs) has been updated, if appropriate
- [ ] [History](https://github.com/CBICA/GaNDLF/blob/master/HISTORY.md) has been updated, if appropriate
- [ ] Tests added or modified to [cover the changes](https://app.codecov.io/gh/CBICA/GaNDLF); if coverage is reduced, please give explanation
- [ ] If customized dependency installation is required (i.e., a separate `pip install` step is needed for PR to be functional), please ensure it is reflected in all the files that control the CI, namely: [python-test.yml](https://github.com/CBICA/GaNDLF/blob/master/.github/workflows/python-test.yml), and all docker files [[1](https://github.com/CBICA/GaNDLF/blob/master/Dockerfile-CPU),[2](https://github.com/CBICA/GaNDLF/blob/master/Dockerfile-CUDA10.2),[3](https://github.com/CBICA/GaNDLF/blob/master/Dockerfile-CUDA11.3),[4](https://github.com/CBICA/GaNDLF/blob/master/Dockerfile-ROCm)]

### CODE REVIEW

**Review by sarthakpati — APPROVED** [review]
LGTM! 

Thoughts, @Geeks-Sid?

**Review by Geeks-Sid — DISMISSED** [review]

File: GANDLF/compute/generic.py [hunk 1]
```diff
@@ -52,10 +53,21 @@ def create_pytorch_objects(parameters, train_csv=None, val_csv=None, device="cpu
...
         val_loader = get_validation_loader(parameters)
 
+    # populate required headers
+    headers_to_populate = headers_to_populate_train
+    if headers_to_populate is None:
+        if headers_to_populate_val is not None:
+            headers_to_populate = headers_to_populate_val
+    if headers_to_populate is not None:
+        parameters = populate_header_in_parameters(parameters, headers_to_populate)
+
```
Comment (sarthakpati) [inline GANDLF/compute/generic.py:70]:
```suggestion
```
We don't need these lines any more

Comment (psfoley) [reply GANDLF/compute/generic.py]: Lines have been removed

**Review by sarthakpati — APPROVED** [review]
LGTM!

**Pull request merged**